File: 프로그래머스/Lv.1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cs
using System;
using System.Collections.Generic;
using System.Linq;
public class Solution {
    public long solution(long n) {
        long answer = 0;
        string answer_s = "";
        List<int> list = new List<int>();
        long x =1;
        int result;
        while(n/x != 0)
        {
            result = (int)((n/x)%10); 
            list.Add(result);
            x = x * 10;
        }
        
        List<int> newList = list.OrderByDescending(i => i).ToList();
        
        for(int i = 0 ; i < newList.Count; i++)
        {
            answer_s = answer_s + newList[i].ToString();
        }
        answer = Int64.Parse(answer_s);
        return answer;
    }
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
class Solution {
    public long solution(long n) {
        int[] count = new int[10];

        while (n > 0)
        {
            int number = (int)(n % 10);
            count[number] += 1;
            n /= 10;
        }
        
        StringBuilder sb = new StringBuilder();
        
        for(int number = count.length - 1; number >= 0; number--)
        {
            if(count[number] == 0)
                continue;
            
            String character = String.valueOf(number);
            
            for(int i = 0; i < count[number]; i++)
            {
                sb.append(character);
            }
        }
        
        long answer = Long.parseLong(sb.toString());
        return answer;
    }
}